connect/runtime/src/test/java/org/apache/kafka/connect/integration/TaskHandle.java

```diff
@@ -149,20 +164,19 @@ public void awaitRecords(long timeoutMillis) throws InterruptedException {
      * @throws InterruptedException if another threads interrupts this one while waiting for records
      */
     public void awaitRecords(long timeout, TimeUnit unit) throws InterruptedException {
-        if (recordsRemainingLatch == null) {
-            throw new IllegalStateException("Illegal state encountered. expectedRecords() was not set for this task?");
-        }
-        if (!recordsRemainingLatch.await(timeout, unit)) {
-            String msg = String.format(
-                    "Insufficient records seen by task %s in %d millis. Records expected=%d, actual=%d",
-                    taskId,
-                    unit.toMillis(timeout),
-                    expectedRecords,
-                    expectedRecords - recordsRemainingLatch.getCount());
-            throw new DataException(msg);
-        }
-        log.debug("Task {} saw {} records, expected {} records",
-                  taskId, expectedRecords - recordsRemainingLatch.getCount(), expectedRecords);
+        recordLatches.awaitRecords(timeout, unit);
+    }
+
+    /**
+     * Wait for this connector to meet the expected number of records as defined by {@code
+     * expectedRecords}.
+     *
+     * @param topic   the name of the topic
+     * @param timeout max duration to wait for records
+     * @throws InterruptedException if another threads interrupts this one while waiting for records
+     */
+    public void awaitRecords(String topic, long timeout) throws InterruptedException {
+        recordLatches.awaitRecords(topic, timeout, TimeUnit.MILLISECONDS);
```

Comment:
Step 6: Added ability to wait until we see the expected # of records _for the specified topic_.